remove experimental v2 from expressions

nathanbegbie opened the pull request:
See https://turnio.slack.com/archives/C02EC0PN15G/p1762153270892709

nathanbegbie committed:
remove experimental v2 from expressions

See https://turnio.slack.com/archives/C02EC0PN15G/p1762153270892709

smn reviewed (dismissed):
Happy! But i suspect somewhere internally in our app we may still be hitting some V2 compat module stuff but we can deal with that when we merge in the version that drops the V2 support.

Please do make sure that when you do release you do a major version bump.

nathanbegbie commented:
@davidkhk just flagging - when I'm back from leave, let's coordinate on this one and getting it in to the `engage` repo

nathanbegbie commented:
https://linear.app/turnio/issue/PROD-4237/remove-expressions-v2-in-breaking-release-of-expressions-lib

nathanbegbie commented:
Once we merge in #305, and we're happy it works, I'll make the breaking change release and get this done. Sorry it's taken so long!

nathanbegbie commented:
I'm now okay with the release process, so will be doing this with the tiny slice of my Friday that can be called calm

nathanbegbie committed:
Version Bump 3.0.0

Removal of these modules consitutes a breaking change, so we're doing a major version bump

nathanbegbie commented:
Just noting that I'm going to bump the version number in this PR as well and do a relase as soon as I merge

smn commented:
Have we confirmed nothing in Turn relies on the Expression.v2 module anywhere?

nathanbegbie commented:
@smn Ah, I was going to make the release, then sort that out downstream in another PR, but probs make sense to do that check before making a blocking release.

Doing that now

smn commented:
well you can tell another repo to just pull the dependency as code from this repo using the `github:` stuff in mix.exs -- may save a bunch of round trips between releases to get an answer to this?

nathanbegbie commented:
> Happy! But i suspect somewhere internally in our app we may still be hitting some V2 compat module stuff but we can deal with that when we merge in the version that drops the V2 support.
> 
> Please do make sure that when you do release you do a major version bump.

@smn hang on a minute, _you_ were the one who suggested that approach ☝️ 😂

nathanbegbie commented:
and in `flow_runner`:

```
% git grep "Expression.V2"
lib/flow_runner.ex:  alias Expression.V2.Compat
lib/flow_runner/custom_blocks/dynamic_select_one_response.ex:  alias Expression.V2.Compat
```

smn commented:
> @smn hang on a minute, _you_ were the one who suggested that approach ☝️ 😂

heh, sorry.
